src/Microsoft.AspNetCore.StaticFiles/DirectoryBrowserMiddleware.cs

```diff
@@ -46,7 +53,7 @@ public DirectoryBrowserMiddleware(RequestDelegate next, IHostingEnvironment host
 
             if (options.Value.Formatter == null)
             {
-                throw new ArgumentException(Resources.Args_NoFormatter);
+                options.Value.Formatter = new HtmlDirectoryFormatter(encoder);
```

Comment:
You should have the formatter as a middleware field instead of mutating the options object. e.g. `_formatter = options.Value.Formatter ?? new HttpDirectoryFormatter(encoder);`